pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go

```diff
@@ -101,125 +92,58 @@ func (tr *TrustedTaskRun) VerifyTask(
-		if err != nil {
-			return apis.ErrGeneric(err.Error(), "spec", "taskRef")
-		}
-		if ts.Name != "" {
-			logger.Info("Verifying TaskRef")
-			if err := VerifyInterface(ctx, &ts.Spec, verifier, signature); err != nil {
-				if err != nil {
-					return apis.ErrGeneric(err.Error(), "spec", "taskRef")
-				}
-			}
-			return nil
-		}
-	}
-
-	return nil
-}
-
-func verifier(
-	ctx context.Context,
-	annotations map[string]string,
-	k8sclient kubernetes.Interface,
-) (signature.Verifier, error) {
-	if annotations[kmsAnnotation] != "" {
-		// Fetch key from kms.
-		return kms.Get(ctx, annotations[kmsAnnotation], crypto.SHA256)
-	} else {
-		cosignPublicKeypath := secretPath
-		// Overwrite the path if set in configmap.
-		cm, err := k8sclient.CoreV1().ConfigMaps(system.Namespace()).Get(ctx, signingConfigMap, metav1.GetOptions{})
-		if err != nil {
-			return nil, err
-		}
-		if cm.Data["signing-secret-path"] != "" {
-			cosignPublicKeypath = cm.Data["signing-secret-path"]
-		}
-		return cosignsignature.LoadPublicKey(ctx, cosignPublicKeypath)
-	}
-}
+	delete(cp.ObjectMeta.Annotations, SignatureAnnotation)
 
-func VerifyInterface(
-	ctx context.Context,
-	taskspec interface{},
-	verifier signature.Verifier,
-	signature []byte,
-) (errs *apis.FieldError) {
-	ts, err := json.Marshal(taskspec)
+	verifier, err := verifier(ctx, cp.ObjectMeta.Annotations, k8sclient)
 	if err != nil {
-		return apis.ErrGeneric(err.Error(), "taskSpec")
+		return apis.ErrGeneric(err.Error(), "metadata")
 	}
 
-	h := sha256.New()
-	h.Write(ts)
+	if err := VerifyInterface(ctx, cp, verifier, signature); err != nil {
+		return apis.ErrGeneric(err.Error(), "taskrun")
+	}
 
-	if err := verifier.VerifySignature(bytes.NewReader(signature), bytes.NewReader(h.Sum(nil))); err != nil {
-		return apis.ErrGeneric(err.Error(), "taskSpec")
+	if tr.Spec.TaskRef != nil {
+		// TODO: add task verification in next PR
 	}
 
 	return nil
 }
 
-func VerifyTaskOCIBundle(
-	ctx context.Context,
-	bundle string,
-	verifier signature.Verifier,
-	signature []byte,
-	k8sclient kubernetes.Interface,
-) (errs *apis.FieldError) {
+// copyTaskRun will exclude system populated fields and return copied taskrun
+func copyTaskRun(in *TrustedTaskRun) (v1beta1.TaskRun, error) {
+	c := v1beta1.TaskRun{}
+	c.TypeMeta = in.TypeMeta
+	c.SetName(in.Name)
+	c.SetGenerateName(in.GenerateName)
+	c.SetNamespace(in.Namespace)
 
-	serviceAccountName := os.Getenv("WEBHOOK_SERVICEACCOUNT_NAME")
-	if serviceAccountName == "" {
-		serviceAccountName = "tekton-verify-task-webhook"
-	}
-	kc, err := k8schain.New(ctx, k8sclient, k8schain.Options{
-		Namespace:          system.Namespace(),
-		ServiceAccountName: serviceAccountName,
-	})
-	if err != nil {
-		return apis.ErrGeneric(err.Error()).ViaKey(bundle)
+	c.Labels = make(map[string]string)
+	for k, v := range in.Labels {
+		c.Labels[k] = v
 	}
+	delete(c.ObjectMeta.Labels, "app.kubernetes.io/managed-by")
```

Comment:
Why exclude this label? 

We should add comments to each field explaining why we're excluding them + add it to the README.

Reply:
it is a label generated by other component not from the user. Will document this